src/System.Private.Xml/src/System/Xml/Schema/CompiledidEntityConstraint.cs

```diff
@@ -56,7 +56,7 @@ public CompiledIdentityConstraint(XmlSchemaIdentityConstraint constraint, XmlNam
             catch (XmlSchemaException e)
             {
                 e.SetSource(constraint.Selector);
-                throw e;
+                throw;
```

Comment:
Yes in general I am all for it! But the SetSource method seems to do something to the caught exception: 

http://source.dot.net/#System.Private.Xml/System/Xml/Schema/XmlSchemaException.cs,209